promql/functions.go

```diff
@@ -184,6 +184,137 @@ func funcIrate(ev *evaluator, args Expressions) model.Value {
 	return resultVector
 }
 
+// Calculate the smoothed value at index i in raw data d.
+// The smoothed value is created by creating two values (x,y) where...
+// x = the current raw data point, weighted by the smoothing factor.
+// y = the last smoothed point + the last trend value weighted by the inverse of the smoothing factor.
+// s = computed smoothed values, b = computed trend factors, d = raw
+func doubleSVal(i int, sf, tf float64, s, b, d []float64) float64 {
+
+	// check for the cached value
+	if f := s[i]; !math.IsNaN(f) {
+		return f
+	}
+
+	x := sf * d[i]
+	y := (1 - sf) * (doubleSVal(i-1, sf, tf, s, b, d) + doubleBVal(i-1, sf, tf, s, b, d))
+
+	// cache value
+	s[i] = x + y
+
+	return s[i]
+}
+
+// Calculate the trend value at the given index i in raw data d.
+// This is somewhat analogous to the slope of the trend at the given index.
+// s = computed smoothed values, b = computed trend factors, d = raw
+func doubleBVal(i int, sf, tf float64, s, b, d []float64) float64 {
+
+	// check for the cached value
+	if f := b[i]; !math.IsNaN(f) {
+		return f
+	}
+
+	x := tf * (doubleSVal(i, sf, tf, s, b, d) - doubleSVal(i-1, sf, tf, s, b, d))
+	y := (1 - tf) * doubleBVal(i-1, sf, tf, s, b, d)
+
+	// cache value
+	b[i] = x + y
+
+	return b[i]
+}
+
+// === holt_winters(node model.ValVector, smoothingFactor model.ValScalar, trendFactor model.ValScalar) Vector ===
+// Holt-Winters is similar to a weighted moving average, where historical data
+// has exponentially less influence on the current data. Holt-Winter also accounts
+// for trends in data. The smoothing factor (0 < sf < 1) effects how historical data will effect the current
+// data. A lower smoothing factor increases the influence of historical data. The trend factor (0 < tf < 1) effects
+// how trends in historical data will effect the current data. A higher trend factor increases the influence
+// of trends.
+// algorithm taken from https://en.wikipedia.org/wiki/Exponential_smoothing titled: "Double exponential smoothing"
+func funcHoltWinters(ev *evaluator, args Expressions) model.Value {
+	mat := ev.evalMatrix(args[0])
+
+	// smoothing factor
```

Comment:
Should I not be matching the style of the comments of the other query functions?